### PR TITLE
[extension-selenium] Add step to wait an element with text matching r…

### PR DESCRIPTION
**Pull request by uarlouski**
…egex

### CODE REVIEW

File: docs/modules/plugins/partials/generic-ui-steps.adoc [hunk 1]
```diff
@@ -415,6 +415,23 @@ Then element located by `$locator` exists for `$duration` duration
 Then element located by `id(banner)` exists for `PT5S` duration
 ----
 
+=== Wait element has text matching regex
```
Comment (valfirst) [inline docs/modules/plugins/partials/generic-ui-steps.adoc:418]:
```suggestion
=== Wait until element has text matching regular expression
```

File: docs/modules/plugins/partials/generic-ui-steps.adoc [hunk 2]
```diff
@@ -415,6 +415,23 @@ Then element located by `$locator` exists for `$duration` duration
...
+
+Waits until an element with the specified locator has text that matches the provided https://www.regular-expressions.info/[regular expression].
+
+[source,gherkin]
+----
+When I wait until element located by `$locator` has text matching `$regex`
+----
+
+* `$locator` - The <<_locator>> of the element whose text to check.
```
Comment (valfirst) [inline docs/modules/plugins/partials/generic-ui-steps.adoc:427]:
```suggestion
* `$locator` - The <<_locator>> of the element which text to check.
```

File: vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java [hunk 3]
```diff
@@ -269,6 +273,36 @@ public void doesElementByLocatorExistsForDuration(Locator locator, Duration dura
...
+                                                                 .map(regex::matcher)
+                                                                 .map(java.util.regex.Matcher::find)
+                                                                 .orElse(false);
+                    }
+
+                    @Override
+                    public String toString()
+                    {
+                        return String.format("text matching regex ('%s') to be present in element located by %s", regex,
+                                locator);
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java:301]: how does passed and failed steps look like?

File: vividus-tests/src/main/resources/story/integration/ScrollStepsTests.story [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 
-Scenario: Scroll RIGHT for element Verify step: When I scroll context to $scrollDirection edge
+Scenario: Scroll RIGHT for element Verify steps: When I scroll context to $scrollDirection edge, When I wait until element located by `$locator` has text matching `$regex`
 Given I am on page with URL `${vividus-test-site-url}/scrollableElements.html`
 When I change context to element located by `id(scrollable)`
 When I scroll context to RIGHT edge
+When I reset context
+When I wait until element located by `id(current-horizontal):a` has text matching `\d+`
 When I change context to element located by `id(current-horizontal):a`
 Then the text matches '\d+'
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/ScrollStepsTests.story:11]:
```suggestion
```

File: vividus-tests/src/main/resources/story/integration/ScrollStepsTests.story [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/ScrollStepsTests.story:12]: also it makes sense to update the test scrolling to bottom

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java (hunk 3 above)
Comment (uarlouski) [reply vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java:301]: Failed:
image

Comment (uarlouski) [reply vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java:301]: shared privately

**Review by valfirst — APPROVED** [review]

**Pull request merged**